src/main/java/org/opensearch/knn/index/VectorDataType.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index;
+
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.document.KnnByteVectorField;
+import org.apache.lucene.document.KnnVectorField;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.IndexableFieldType;
+import org.apache.lucene.index.VectorEncoding;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.opensearch.index.mapper.ParametrizedFieldMapper;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.opensearch.knn.common.KNNConstants.DEFAULT_VECTOR_DATA_TYPE;
+import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
+import static org.opensearch.knn.common.KNNConstants.LUCENE_NAME;
+import static org.opensearch.knn.common.KNNConstants.VECTOR_DATA_TYPE;
+
+/**
+ * Enum contains data_type of vectors and right now only supported for lucene engine in k-NN plugin.
+ * We have two vector data_types, one is float (default) and the other one is byte.
+ */
+public enum VectorDataType {
+    BYTE("byte") {
+        /**
+         * @param dimension  Dimension of the vector
+         * @param vectorSimilarityFunction VectorSimilarityFunction for a given spaceType
+         * @return FieldType of type KnnByteVectorField
+         */
+        @Override
+        public FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction) {
+            return KnnByteVectorField.createFieldType(dimension, vectorSimilarityFunction);
+        }
+
+        /**
+         * @param knnEngine KNNEngine
+         * @return DocValues FieldType of type Binary and with BYTE VectorEncoding
+         */
+        @Override
+        public FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
+            IndexableFieldType indexableFieldType = new IndexableFieldType() {
```

Comment:
plus why we need all this? for Float the same function is very simple. Anything I am missing here.

Reply:
Looking at the VectorEncoding field in the FieldType, initially I thought that I might need to set the VectorEncoding also to BYTE and there is no setter method to directly set the VectorEncoding to a given type. We have a method [setVectorAttributes](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/document/FieldType.java#L375) to set VectorEncoding but we must set the dimension also to a non zero value. For docValues we shouldn't set the dimension, if we set it we will run into an exception because it will also [index](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexingChain.java#L728-L730) that binary value as a vector. The alternative way is to override this IndexableFiledType and set the Encoding.

But, based on my reply to above question I doubt and need confirmation from team that we may not ned to make any new changes for building docValuesFieldType. So, can you confirm ?